### PR TITLE
Make sure Q is in a real Node environment

### PR DESCRIPTION
**Pull request by kahnjw**
This adds an additional check to ensure that the `process` global Q looks for is closer to the global that nodejs environments expose. This is in light of updates to `node-process` a process shim used by Browserify:

https://github.com/defunctzombie/node-process/

This library no longer uses `setImmediate` if it is available, it uses `setTimeout` out of the box. Because Browserify is getting so popular as a moduling system, I think it is wise to add this check, so Browserify user's get all the extra `setImmediate` juice they possibly can.

By default, Browserify will scan files for `process` usage and insert the global as needed. Most users won't read this, and if they do, they probably won't realize there is a pretty big performance penalty if this setting is left on. Turning it off will also mean that certain modules will no longer work. I think this solution is the best of both worlds: get the added performance and still use the `process` global if needed.

>When opts.detectGlobals is true, scan all files for process, global, __filename, and __dirname, defining as necessary. With this option npm modules are more likely to work but bundling takes longer. Default true.

The solution I proposed is really simple, but it does the t(r)ick. `toString`ing the `process` global in node yields: `'[object process]'`, while the `process` shim Browserify uses yields: `'[object Object]'`.

The module [detect-node](https://github.com/iliakan/detect-node) uses a similar method.

### CODE REVIEW

File: q.js [hunk 1]
```diff
@@ -155,9 +155,15 @@ var nextTick =(function () {
         }
     };
 
-    if (typeof process !== "undefined" && process.nextTick) {
+    if (typeof process === "object" && (process + '') === '[object process]' &&
```
Comment (kriskowal) [inline q.js]: Let’s toString explicitly.

**Comment by kriskowal** [comment]
This looks good to me. I will land and release this by end of weekend and hopefully sooner.

**Comment by kahnjw** [comment]
@kriskowal done.

**Comment by kriskowal** [comment]
This is released in v1.2.1. Thanks for looking into this.

**Pull request merged**